flake: TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error

https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/kcp-dev_kcp/2747/pull-ci-kcp-dev-kcp-main-e2e-sharded/1623282406905614336

```
=== RUN   TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
=== PAUSE TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    assertions.go:1691: Waiting for condition, but got: workspace phase is Initializing, not Ready
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1928: Created root:organization workspace root:e2e-workspace-rr7tp as /clusters/1frk2srbeaqtnxsu on shard "root"
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1930: Created root:universal workspace root:e2e-workspace-rr7tp:upsyncer as /clusters/2d8bqqfioo3e85eh on shard "root"
    virtualworkspace_test.go:1933: [11:53:56.179579] Deploying syncer into workspace root:e2e-workspace-rr7tp:upsyncer
    virtualworkspace_test.go:1951: Saving test artifacts for test "TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error." under "/logs/artifacts/TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error./387534887".
    virtualworkspace_test.go:1951: Configuring workspace root:e2e-workspace-rr7tp:upsyncer for syncing
    virtualworkspace_test.go:1951: running: KUBECONFIG=/logs/artifacts/TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error./387534887/artifacts/root_e2e-workspace-rr7tp_upsyncer.kubeconfig kubectl kcp workload sync upsyncer --syncer-image=not-a-valid-image --output-file=- --qps=-1 --feature-gates=CustomResourceValidationExpressions=true --api-import-poll-interval=5s --downstream-namespace-clean-delay=2s --resources=persistentvolumes --apiexports=
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1951: Created root:universal workspace root:e2e-workspace-rr7tp:upsyncer:upsyncer-sink as /clusters/1qwl68qxlu4fz8mw on shard "shard-1"
I0208 11:54:01.112791   31982 bootstrap.go:129] "bootstrapping CRD" crd="endpoints.core"
I0208 11:54:01.185381   31982 bootstrap.go:129] "bootstrapping CRD" crd="services.core"
I0208 11:54:01.188634   31982 bootstrap.go:129] "bootstrapping CRD" crd="ingresses.networking.k8s.io"
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1945: [11:54:08.559638] Installing test CRDs into sink cluster...
I0208 11:54:08.594667   31982 bootstrap.go:129] "bootstrapping CRD" crd="services.core"
I0208 11:54:08.676597   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="2mzrkl6txw854u3l" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.711839   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="24b3isjk53c5a4ph" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.718740   31982 bootstrap.go:150] "bootstrapped CRD" crd="services.core" customresourcedefinition.workspace="2a196tlapppplwf9" customresourcedefinition.namespace="" customresourcedefinition.name="services.core" customresourcedefinition.apiVersion="" duration="668.213408ms"
I0208 11:54:08.718816   31982 bootstrap.go:169] "waiting for CRD to be established" crd="services.core" customresourcedefinition.workspace="2a196tlapppplwf9" customresourcedefinition.namespace="" customresourcedefinition.name="services.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.735884   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="services.core" customresourcedefinition.workspace="24b3isjk53c5a4ph" customresourcedefinition.namespace="" customresourcedefinition.name="services.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.750975   31982 bootstrap.go:150] "bootstrapped CRD" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="5navvzd28lex62ps" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion="" duration="203.027809ms"
I0208 11:54:08.751059   31982 bootstrap.go:169] "waiting for CRD to be established" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="5navvzd28lex62ps" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.751129   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="pods.core" customresourcedefinition.workspace="2b2k63sllfjyn5sx" customresourcedefinition.namespace="" customresourcedefinition.name="pods.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.757688   31982 bootstrap.go:150] "bootstrapped CRD" crd="endpoints.core" customresourcedefinition.workspace="5navvzd28lex62ps" customresourcedefinition.namespace="" customresourcedefinition.name="endpoints.core" customresourcedefinition.apiVersion="" duration="240.672193ms"
I0208 11:54:08.757766   31982 bootstrap.go:169] "waiting for CRD to be established" crd="endpoints.core" customresourcedefinition.workspace="5navvzd28lex62ps" customresourcedefinition.namespace="" customresourcedefinition.name="endpoints.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.763611   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="2a196tlapppplwf9" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.765270   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="endpoints.core" customresourcedefinition.workspace="24b3isjk53c5a4ph" customresourcedefinition.namespace="" customresourcedefinition.name="endpoints.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.770892   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="endpoints.core" customresourcedefinition.workspace="2a196tlapppplwf9" customresourcedefinition.namespace="" customresourcedefinition.name="endpoints.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.818642   31982 bootstrap.go:187] "CRD has no Established condition" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="5navvzd28lex62ps" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.836737   31982 bootstrap.go:187] "CRD is not established: Installing: the initial names have been accepted" crd="services.core" customresourcedefinition.workspace="2a196tlapppplwf9" customresourcedefinition.namespace="" customresourcedefinition.name="services.core" customresourcedefinition.apiVersion=""
I0208 11:54:08.865036   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="ingresses.networking.k8s.io" customresourcedefinition.workspace="24b3isjk53c5a4ph" customresourcedefinition.namespace="" customresourcedefinition.name="ingresses.networking.k8s.io" customresourcedefinition.apiVersion=""
I0208 11:54:08.865239   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="endpoints.core" customresourcedefinition.workspace="24b3isjk53c5a4ph" customresourcedefinition.namespace="" customresourcedefinition.name="endpoints.core" customresourcedefinition.apiVersion=""
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1951: running: kubectl --kubeconfig /logs/artifacts/TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error./387534887/artifacts/root_e2e-workspace-rr7tp_upsyncer_upsyncer-sink.kubeconfig apply -f -
I0208 11:54:09.737867   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="persistentvolumes.core" customresourcedefinition.workspace="3427yiy16lzput9l" customresourcedefinition.namespace="" customresourcedefinition.name="persistentvolumes.core" customresourcedefinition.apiVersion=""
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    syncer.go:726: checking secret kcp-syncer-upsyncer-2yqr8cfh/default-token-sqh2d for annotation kubernetes.io/service-account.name=kcp-syncer-upsyncer-2yqr8cfh
    syncer.go:726: checking secret kcp-syncer-upsyncer-2yqr8cfh/kcp-syncer-upsyncer-2yqr8cfh for annotation kubernetes.io/service-account.name=kcp-syncer-upsyncer-2yqr8cfh
    syncer.go:726: checking secret kcp-syncer-upsyncer-2yqr8cfh/kcp-syncer-upsyncer-2yqr8cfh-token for annotation kubernetes.io/service-account.name=kcp-syncer-upsyncer-2yqr8cfh
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1951: Cluster "upsyncer" is Ready
    virtualworkspace_test.go:1953: [11:54:14.905599] Bind upsyncer workspace
    virtualworkspace_test.go:1956: Bind workload workspace root:e2e-workspace-rr7tp:upsyncer
    virtualworkspace_test.go:1956: running: KUBECONFIG=/logs/artifacts/TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error./387534887/artifacts/root_e2e-workspace-rr7tp_upsyncer.kubeconfig kubectl kcp bind compute root:e2e-workspace-rr7tp:upsyncer --apiexports=root:e2e-workspace-rr7tp:upsyncer:kubernetes
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1958: [11:54:15.956698] Waiting for the persistentvolumes crd to be imported and available in the upsyncer source cluster...
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.082938] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.197880] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
I0208 11:54:16.218734   31982 bootstrap.go:187] "CRD is not established: InitialNamesAccepted: the initial names have been accepted" crd="persistentvolumes.core" customresourcedefinition.workspace="2b2k63sllfjyn5sx" customresourcedefinition.namespace="" customresourcedefinition.name="persistentvolumes.core" customresourcedefinition.apiVersion=""
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.286667] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.379079] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
I0208 11:54:16.447861   31982 discovery.go:186] "processing discovery" resources=[persistentvolumes] reconciler="api-importer" groupVersion="v1" resource="persistentvolumes" kind="PersistentVolume" crd="persistentvolumes.core"
I0208 11:54:16.471859   31982 bootstrap.go:129] "bootstrapping CRD" crd="persistentvolumes.core"
I0208 11:54:16.475816   31982 apiimporter.go:320] "creating APIResourceImport" resources=[] reconciler="api-importer" group="" version="v1" resource="pods"
I0208 11:54:16.509800   31982 apiimporter.go:320] "creating APIResourceImport" resources=[persistentvolumes] reconciler="api-importer" group="" version="v1" resource="persistentvolumes"
    virtualworkspace_test.go:1962: [11:54:16.515361] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.677032] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.704627] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.797590] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.869398] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:16.968343] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:17.092839] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
W0208 11:54:17.095766   31982 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:17.237366] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:17.321716] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:17.412291] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1962: [11:54:17.534957] error seen waiting for persistentvolumes crd to become active: the server could not find the requested resource
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1970: [11:54:18.368988] Starting test...
    virtualworkspace_test.go:1971: [11:54:18.370205] Creating a persistentvolume in the kubelike source cluster...
    virtualworkspace_test.go:1971: [11:54:18.370339] Creating PV test-pv through direct kube client ...
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1971: [11:54:20.388468] Getting PV test-pv through upsyncer virtual workspace...
=== CONT  TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    virtualworkspace_test.go:1971: 
        	Error Trace:	virtualworkspace_test.go:1858
        	            				virtualworkspace_test.go:1971
        	Error:      	Received unexpected error:
        	            	the server could not find the requested resource (get persistentvolumes test-pv)
        	Test:       	TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error.
    syncer.go:216: Collecting imported resource info: /logs/artifacts/TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error./387534887/artifacts
I0208 11:54:20.407079   31982 apiimporter.go:186] "stopping API Importer" resources=[persistentvolumes] reconciler="api-importer"
I0208 11:54:20.407144   31982 reflector.go:277] Stopping reflector *v1alpha1.SyncTarget (10h0m0s) from k8s.io/client-go@v0.0.0-20230127203538-f48a2e575b3b/tools/cache/reflector.go:215
E0208 11:54:20.407181   31982 request.go:977] Unexpected error when reading response body: context canceled
E0208 11:54:20.407321   31982 apiimporter.go:273] "error updating APIResourceImport" err="unexpected error when reading response body. Please retry. Original error: context canceled" resources=[persistentvolumes] reconciler="api-importer" group="" version="v1" resource="persistentvolumes" apiResourceImport="persistentvolumes.upsyncer.v1.core"
I0208 11:54:20.407081   31982 reflector.go:277] Stopping reflector *v1alpha1.APIResourceImport (10h0m0s) from k8s.io/client-go@v0.0.0-20230127203538-f48a2e575b3b/tools/cache/reflector.go:215
I0208 11:54:20.425298   31982 shared_informer.go:312] caches populated
I0208 11:54:20.425686   31982 apiimporter.go:174] "Starting API Importer" resources=[persistentvolumes] reconciler="api-importer"
I0208 11:54:20.425790   31982 apiimporter.go:227] "Importing APIs" resources=[persistentvolumes] reconciler="api-importer" resourcesToImport=[persistentvolumes]
    --- FAIL: TestUpsyncerVirtualWorkspace/update_a_persistentvolume_in_kcp_through_upsyncer_virtual_workspace,_try_to_remove_the_upsync_state_label,_expect_error. (40.55s)
```

--- a/test/e2e/virtual/syncer/virtualworkspace_test.go
+++ b/test/e2e/virtual/syncer/virtualworkspace_test.go
@@ -1956,13 +1956,24 @@ func TestUpsyncerVirtualWorkspace(t *testing.T) {
 			).Bind(t)
 
 			logWithTimestampf(t, "Waiting for the persistentvolumes crd to be imported and available in the upsyncer source cluster...")
-			require.Eventually(t, func() bool {
+			framework.Eventually(t, func() (bool, string) {
 				_, err := kubeClusterClient.CoreV1().PersistentVolumes().Cluster(upsyncerPath).List(ctx, metav1.ListOptions{})
 				if err != nil {
-					logWithTimestampf(t, "error seen waiting for persistentvolumes crd to become active: %v", err)
-					return false
+					return false, err.Error()
 				}
-				return true
+				return true, ""
+			}, wait.ForeverTestTimeout, time.Millisecond*100)
+
+			kubelikeUpsyncerVWClient, err := kcpkubernetesclientset.NewForConfig(upsyncer.UpsyncerVirtualWorkspaceConfig)
+			require.NoError(t, err)
+
+			logWithTimestampf(t, "Waiting for the persistentvolumes to be available in the upsyncer virtual workspace...")
+			framework.Eventually(t, func() (bool, string) {
+				_, err := kubelikeUpsyncerVWClient.CoreV1().PersistentVolumes().Cluster(upsyncerClusterName.Path()).List(ctx, metav1.ListOptions{})
+				if err != nil {
+					return false, err.Error()
+				}
+				return true, ""
 			}, wait.ForeverTestTimeout, time.Millisecond*100)
 
 			syncTargetKey := upsyncer.ToSyncTargetKey()

